staging/src/k8s.io/apiserver/pkg/server/serve.go

```diff
@@ -84,32 +83,34 @@ func (s *GenericAPIServer) serveSecurely(stopCh <-chan struct{}) error {
 	}
 
 	glog.Infof("Serving securely on %s", s.SecureServingInfo.BindAddress)
-	var err error
-	s.effectiveSecurePort, err = RunServer(secureServer, s.SecureServingInfo.BindNetwork, s.ShutdownTimeout, stopCh)
+	err := RunServer(secureServer, s.SecureServingInfo.Listener, s.SecureServingInfo.BindNetwork, s.ShutdownTimeout, stopCh)
```

Comment:
The BindNetwork support here is too late for downstream OpenShift. Can we move that into the SecureServingOptions as well and require `ln != nil` in `RunServer`. I think that would make the whole logic simpler as we have one code path less: the config SecureServingInfo must always have a valid listener.